token/cli/src/config.rs

```diff
@@ -133,36 +133,37 @@ impl<'a> Config<'a> {
 
         let default_keypair = cli_config.keypair_path.clone();
 
-        let default_signer: Arc<dyn Signer> = {
+        let default_signer: Option<Arc<dyn Signer>> = {
             if let Some(owner_path) = matches.value_of("owner") {
-                signer_from_path_with_config(matches, owner_path, "owner", wallet_manager, &config)
+                signer_from_path_with_config(matches, owner_path, "owner", wallet_manager, &config).ok()
             } else {
-                signer_from_path_with_config(
-                    matches,
-                    &default_keypair,
-                    "default",
-                    wallet_manager,
-                    &config,
-                )
+                resolve_signer_from_path(matches, &default_keypair, "default", wallet_manager)
+                    .unwrap_or_else(|e| {
+                        eprintln!("error: {}", e);
+                        exit(1);
+                    }).map(|path| read_keypair_file(&path)
+                                .unwrap_or_else(|e| {
+                                    eprintln!("error: {}", e);
+                                    exit(1);
+                                }).into())
```

Comment:
I think I've made the change you requested.  I didn't resolve this conversation because I was waiting to hear your confirmation that it was OK now.  Thanks.

Reply:
Since `read_keypair_file` only works with keypair files, and not ledgers, you'll need to essentially throw away the result of `resolve_signer_from_path` and still use `signer_from_path_with_config`.

The only use of `resolve_signer_from_path` is to check that there's something there, meaning there's just this additional change to make:

```suggestion
                    }).map(|_path| signer_from_path_with_config(matches, &default_keypair, "default", wallet_manager, &config)
                                .unwrap_or_else(|e| {
                                    eprintln!("error: {}", e);
                                    exit(1);
                                }).into())
```

Does that make sense?